core/src/js/services/mercenaries/parser/mercenaries-ability-revealed-parser.ts

```diff
@@ -27,7 +27,7 @@ export class MercenariesAbilityRevealedParser implements MercenariesParser {
 		const isPlayer = controllerId === localPlayer.PlayerId;
 		const team = isPlayer ? battleState.playerTeam : battleState.opponentTeam;
 		const abilityOwner = team.getMercenary(ownerEntityId);
-		if (!cardId) {
+		if (!cardId || battleState.currentTurn!=0) {
```

Comment:
I'm not totally sure about this change. When an ability swap from one to another (which is not yet supported), we might have to handle it in this parser (along with an "ability_hidden" event maybe). 
Also, why hide abilities revealed during the fight in the first place? I'm not too familiar with this use case, why aren't they valid?